File: JavaScript/6js 1/expandedForm.js
/*
|--------------------------------------------------------------------------
| Write Number in Expanded Form
|--------------------------------------------------------------------------
|
*/

function expandedForm(num) {
   let res = [];
   let x = num.toString();
   for (let i = 0; i < x.length; i++) {
      if (x[i] !== "0") res.push(x[i] + "0".repeat(x.length - i - 1));
   }
   return res.join(" + ");
}


function expandedForm(num) {
   return `${num}`
      .replace(/./g, (dgt, idx, str) => `${dgt * 10 ** (str.length - ++idx)} + `)
      .replace(/\D+(0|$)/g, "");
}


function expandedForm(num) {
   return String(num)
      .split("")
      .map((num, index, arr) => num + "0".repeat(arr.length - index - 1))
      .filter((num) => Number(num) != 0)
      .join(" + ");
}


console.log(expandedForm(12), "10 + 2");
console.log(expandedForm(42), "40 + 2");
console.log(expandedForm(70304), "70000 + 300 + 4");

File: Python/6py 1/expanded_form.py
"""
|--------------------------------------------------------------------------
| Write Number in Expanded Form
|--------------------------------------------------------------------------
|
"""

def expanded_form(num):
   res = []
   for i in range(len(str(num))):
      if str(num)[i] != "0": res.append(str(num)[i] + "0" * (len(str(num)) - i - 1))
   return " + ".join(res)


def expanded_form(num):
   return ' + '.join([x+'0'*i for i,x in enumerate(str(num)[::-1]) if x != '0'][::-1])


def expanded_form(num):
   num = str(num)
   st = ''
   for j, i in enumerate(num):
      if i != '0':
         st += ' + {}{}'.format(i, (len(num[j+1:])*'0'))
   return st.strip(' +')


print(expanded_form(12), '10 + 2')
print(expanded_form(42), '40 + 2')
print(expanded_form(70304), '70000 + 300 + 4')